internal/service/codebuild/fleet.go

```diff
@@ -0,0 +1,598 @@
+	var diags diag.Diagnostics
+	conn := meta.(*conns.AWSClient).CodeBuildClient(ctx)
+
+	input := &codebuild.UpdateFleetInput{
+		Arn: aws.String(d.Id()),
+	}
+
+	if d.HasChange("base_capacity") {
+		input.BaseCapacity = aws.Int32(int32(d.Get("base_capacity").(int)))
+	}
+
+	if d.HasChange("compute_type") {
+		input.ComputeType = types.ComputeType(d.Get("compute_type").(string))
+	}
+
+	if d.HasChange("environment_type") {
+		input.EnvironmentType = types.EnvironmentType(d.Get("environment_type").(string))
+	}
+
+	if d.HasChange("fleet_service_role") {
+		input.FleetServiceRole = aws.String(d.Get("fleet_service_role").(string))
+	}
+
+	if d.HasChange("image_id") {
+		input.ImageId = aws.String(d.Get("image_id").(string))
+	}
+
+	// Make sure that overflow_behavior is set (if defined) on update - API omits it on updates.
+	if v, ok := d.GetOk("overflow_behavior"); ok {
+		input.OverflowBehavior = types.FleetOverflowBehavior(v.(string))
+	}
+
+	if d.HasChange("scaling_configuration") {
+		if v, ok := d.GetOk("scaling_configuration"); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
+			input.ScalingConfiguration = expandScalingConfiguration(v.([]interface{})[0].(map[string]interface{}))
+		}
+	}
+
+	if d.HasChange(names.AttrVPCConfig) {
+		if v, ok := d.GetOk(names.AttrVPCConfig); ok && len(v.([]interface{})) > 0 && v.([]interface{})[0] != nil {
+			input.VpcConfig = expandVPCConfig(v.([]interface{})[0].(map[string]interface{}))
+		}
+	}
+
+	input.Tags = getTagsIn(ctx)
+
+	log.Printf("[DEBUG] Updating CodeBuild Fleet (%s): %#v", d.Id(), input)
+
+	_, err := tfresource.RetryWhenIsAErrorMessageContains[*types.InvalidInputException](ctx, propagationTimeout, func() (interface{}, error) {
+		return conn.UpdateFleet(ctx, input)
+	}, "ot authorized to perform")
+
+	if err != nil {
+		return create.AppendDiagError(diags, names.CodeBuild, create.ErrActionUpdating, ResNameFleet, d.Id(), err)
+	}
+
+	if err := waitFleetActive(ctx, conn, d.Id()); err != nil {
+		return create.AppendDiagError(diags, names.CodeBuild, create.ErrActionWaitingForUpdate, ResNameFleet, d.Id(), err)
+	}
+
+	return append(diags, resourceFleetRead(ctx, d, meta)...)
+}
+
+func resourceFleetDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	conn := meta.(*conns.AWSClient).CodeBuildClient(ctx)
+
+	log.Printf("[INFO] Deleting CodeBuild Fleet %s", d.Id())
+
+	_, err := conn.DeleteFleet(ctx, &codebuild.DeleteFleetInput{
+		Arn: aws.String(d.Id()),
+	})
+
+	if errs.IsA[*types.ResourceNotFoundException](err) {
+		return diags
+	}
+	if err != nil {
+		return create.AppendDiagError(diags, names.CodeBuild, create.ErrActionDeleting, ResNameFleet, d.Id(), err)
+	}
+
+	if err := waitFleetDeleted(ctx, conn, d.Id()); err != nil {
+		return create.AppendDiagError(diags, names.CodeBuild, create.ErrActionWaitingForDeletion, ResNameFleet, d.Id(), err)
+	}
```

Comment:
Fleets can stay in the "pending deletion" state even for a few hours. Not sure what would be the best way to handle such cases (e.g. allow optionally skipping the wait?).

Reply:
Can fleets be "undeleted"? I not we may just want to treat "pending deletion" as deleted

Reply:
Apparently it cannot. When I try to perform any action on a fleet "pending deletion", I get the following error:
```
Fleet status is PENDING_DELETION_SUBMITTED. Cannot be updated.
```

Therefore, I'll remove the waiter.

Reply:
@ewbankkit Done ✅